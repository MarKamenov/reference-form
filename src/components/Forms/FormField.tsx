
export interface FormFieldProps {
  id: string;
  name: string;
  label: string;
  type: string
  handleChange: (ev: React.ChangeEvent<HTMLInputElement>) => void
  value: string;
  required?: boolean
}

export const FormField = ({ id, label, name, type, handleChange, value, required }: FormFieldProps) => {

  return (
    <>
      <label htmlFor={id}>{label}</label>
      <input
        data-testid="input"
        id={id}
        name={name}
        type={type}
        value={value}
        onChange={(e) => handleChange(e)}
        required={required}
      />
    </>
  );
};
