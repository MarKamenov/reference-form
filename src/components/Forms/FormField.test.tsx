import { render } from '@testing-library/react';
import { screen } from '@testing-library/dom'

import { FormField } from './FormField';

describe('FormField component', () => {
    const mockedOnChange = jest.fn();

    test('should render component', () => {
        const { container } = render(<FormField id='1' name='form-name' label='form label' type='text' value='text' required handleChange={mockedOnChange} />)

        expect(container).toBeInTheDocument();
    });

    test('should match snapshot', () => {
        const { container } = render(<FormField id='1' name='form-name' label='form label' type='text' value='text' required handleChange={mockedOnChange} />);

        expect(container).toMatchSnapshot();
    });

    test('should render the label', () => {
        const label = 'form label'
        render(<FormField id='1' name='form-name' label='form label' type='text' value='text' required handleChange={mockedOnChange} />)

        expect(screen.getByText(label)).toBeInTheDocument()
    });

    test('should find the input field', () => {
        render(<FormField id='1' name='form-name' label='form label' type='text' value='text' required handleChange={mockedOnChange} />)

        const input: HTMLInputElement = screen.getByTestId('input')

        expect(input.value).toBe('text')
    });
});

