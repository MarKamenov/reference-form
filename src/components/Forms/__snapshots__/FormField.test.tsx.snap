// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField component should match snapshot 1`] = `
<div>
  <label
    for="1"
  >
    form label
  </label>
  <input
    data-testid="input"
    id="1"
    name="form-name"
    required=""
    type="text"
    value="text"
  />
</div>
`;
